pkg/log/options.go

```diff
@@ -0,0 +1,55 @@
+package log
+
+import (
+	"flag"
+	"fmt"
+
+	json "k8s.io/component-base/logs/json"
+	"k8s.io/klog/v2"
+)
+
+const (
+	logFormatFlagName = "log-format"
+	defaultLogFormat  = "text"
+	jsonLogFormat     = "json"
+)
+
+// Options has klog format parameters.
+type Options struct {
+	LogFormat string
+}
+
+// NewOptions return new klog options.
+func NewOptions() *Options {
+	return &Options{
+		LogFormat: defaultLogFormat,
+	}
+}
+
+// AddFlags adds log-format flag.
+func (o *Options) AddFlags() {
+	fs := flag.CommandLine
+	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf("Sets the logging format. One of (%s|%s)", defaultLogFormat, jsonLogFormat))
+}
+
+// Validate validates the log-format flag.
+func (o *Options) Validate() error {
+	if o.LogFormat != defaultLogFormat && o.LogFormat != jsonLogFormat {
+		return fmt.Errorf("unknown logging format %s. Only \"%s\" and \"%s\" are supported", o.LogFormat, defaultLogFormat, jsonLogFormat)
+	}
+
+	return nil
+}
+
+// Apply set klog logger from LogFormat type and set log severity to appropriate io.Writer.
```

Comment:
Is this comment accurate? We're not setting log severity to the appropriate `io.Writer`

Reply:
Removed inaccurate comments.